doc/dev/sharp_bits.rst

```diff
@@ -755,6 +755,108 @@ a single RX gate is being applied due to the rotation gate merger:
 
 Note that currently PennyLane transforms **cannot** be applied when ``autograph=True``.
 
+Compatibility with PennyLane decompositions
+-------------------------------------------
+
+When defining decompositions of PennyLane operations, any control flow depending on dynamic
+variables will fail, since decompositions are applied at compile time:
+
+.. code-block:: python
+
+    class RXX(qml.operation.Operation):
+        num_params = 1
+        num_wires = 2
+
+        def compute_decomposition(self, *params, wires=None):
+            theta = params[0]
+            ops = []
+
+            if theta == 0.3:
+                ops.append(qml.PauliRot(theta / 2 * 2, 'XX', wires=wires))
+            else:
+                ops.append(qml.PauliRot(theta / 2 * 2, 'XX', wires=wires))
+
+            return ops
+
+    dev = qml.device("lightning.qubit", wires=2)
+
+    @qjit
+    @qml.qnode(dev)
+    def circuit(theta):
+        RXX(theta, wires=[0, 1])
+        qml.Hadamard(1)
+        return qml.expval(qml.PauliZ(0))
+
+>>> circuit(0.3)
+TracerBoolConversionError: Attempted boolean conversion of traced array with shape bool[]..
+See https://jax.readthedocs.io/en/latest/errors.html#jax.errors.TracerBoolConversionError
+
+Instead, Catalyst control flow must be used in order to support control flow on dynamic variables:
```

Comment:
@josh146 would you like to add a reference to the list of control flow operations? 

https://docs.pennylane.ai/projects/catalyst/en/stable/dev/quick_start.html#compiling-with-control-flow

Reply:
Hmm I might not, since I don't want to encode `stable`/`latest` into the URL. I think users if reading the sharp bits guide can likely navigate to the control functions

Reply:
```suggestion
Instead, Catalyst control flow (such as :func:`~.cond` and :func:`.for_loop`) must be used in order to support control flow on dynamic variables:
```